How do I keep 2 lines after functions when piping black?

I'm loving jupytext and the ability to keep my code clean and lint-free! I'm just trying to get my workflow as clean as possible and running into one little snag. `flake8` wants to have 2 empty lines after functions, but when piping `black` via jupytext, those 2 lines don't get preserved for me.

# How to replicate:

1. Create a file called `notebook.py` with the following content:
    ```py
    # %%
    def func():
        return 42
    # %%
    func()
    ```
2. Create a synced `.ipynb` file:
    ```shell
    $ jupytext --set-formats ipynb,py:percent --sync notebook.py
    [jupytext] Reading notebook.py in format py
    [jupytext] Updating notebook metadata with '{"jupytext": {"formats": "ipynb,py:percent"}}'
    [jupytext] Executing black -
    All done! ✨ 🍰 ✨
    1 file left unchanged.
    [jupytext] Updating notebook.ipynb
    [jupytext] Updating notebook.py
    $ cat notebook.py
    # ---
    # jupyter:
    #   jupytext:
    #     cell_metadata_filter: -all
    #     formats: ipynb,py:percent
    #     text_representation:
    #       extension: .py
    #       format_name: percent
    #       format_version: '1.3'
    #       jupytext_version: 1.10.3
    # ---
    
    # %%
    def func():
        return 42
    # %%
    func()
    $ cat notebook.ipynb
    {
     "cells": [
      {
       "cell_type": "code",
       "execution_count": null,
       "id": "8e0e0c0c",
       "metadata": {
        "lines_to_next_cell": 0
       },
       "outputs": [],
       "source": [
        "def func():\n",
        "    return 42"
       ]
      },
      {
       "cell_type": "code",
       "execution_count": null,
       "id": "745567b8",
       "metadata": {},
       "outputs": [],
       "source": [
        "func()"
       ]
      }
     ],
     "metadata": {
      "jupytext": {
       "cell_metadata_filter": "-all",
       "formats": "ipynb,py:percent",
       "main_language": "python"
      }
     },
     "nbformat": 4,
     "nbformat_minor": 5
    }
    ```
3. Lint file to realize that `flake8` wants 2 spaces after functions:
    ```
    $ flake8 notebook.py
    notebook.py:17:1: E305 expected 2 blank lines after class or function definition, found 0
    ```
4. Run black on the .py file to add the 2 spaces back in, satisfying flake8:
    ```
    $ black notebook.py
    reformatted notebook.py
    All done! ✨ 🍰 ✨
    1 file reformatted.
    $ flake8 notebook.py
    $
    ```
5. Pipe black through jupytext, which gets rid of those 2 lines again:
    ```
    $ jupytext --sync --pipe black notebook.py
    [jupytext] Reading notebook.py in format py
    [jupytext] Loading notebook.ipynb
    [jupytext] Executing black -
    reformatted -
    All done! ✨ 🍰 ✨
    1 file reformatted.
    [jupytext] Updating notebook.py
    $ flake8 notebook.py
    notebook.py:17:1: E305 expected 2 blank lines after class or function definition, found 0
    ```
6. Manually change the `lines_to_next_cell` metadata from `0` to `2` in the notebook.ipynb, then repeat:
    ```
    $ jupytext --sync --pipe black notebook.py
    [jupytext] Reading notebook.py in format py
    [jupytext] Loading notebook.ipynb
    [jupytext] Executing black -
    All done! ✨ 🍰 ✨
    1 file left unchanged.
    [jupytext] Updating notebook.py
    $ flake8 notebook.py
    $
    ```

It seems to me that the problem is that the `lines_to_next_cell` metadata is sticky, and changing the lines between cells in the python file doesn't affect that value after the cell is initially created. Is that something that could change? Could you make it so that when the `.py` file is newer, the `lines_to_next_cell` value gets updated in the notebook if there are changes, or would that break other functionality? And if that would break things, is there another solution for this kind of situation besides turning off that particular error in flake8, which will be my current solution?

--- a/docs/CHANGELOG.md
+++ b/docs/CHANGELOG.md
@@ -10,6 +10,8 @@ Jupytext ChangeLog
 
 **Fixed**
 - All text files are opened with an explicit `utf-8` encoding ([#770](https://github.com/mwouts/jupytext/issues/770))
+- Previously `--pipe black` was not always putting two blank lines between functions. To fix that we load the internal Jupytext
+  cell metadata like `lines_to_next_cell` from the text file rather than ipynb ([#761](https://github.com/mwouts/jupytext/issues/761))
 
 
 1.11.1 (2021-03-26)

--- a/jupytext/metadata_filter.py
+++ b/jupytext/metadata_filter.py
@@ -215,6 +215,8 @@ def restore_filtered_metadata(
     metadata = copy(filtered_metadata)
     for key in unfiltered_metadata:
         if key not in filtered_unfiltered_metadata:
-            metadata[key] = unfiltered_metadata[key]
+            # We don't want to restore the line_to_next_cell metadata from the ipynb file, see #761
+            if key not in _JUPYTEXT_CELL_METADATA:
+                metadata[key] = unfiltered_metadata[key]
 
     return metadata

--- a/tests/test_black.py
+++ b/tests/test_black.py
@@ -228,3 +228,27 @@ def test_black_through_tempfile(
     jupytext([tmp_md, "--pipe", "black {}"])
     with open(tmp_md) as fp:
         compare(fp.read(), black)
+
+
+@requires_black
+def test_pipe_black_removes_lines_to_next_cell_metadata(
+    tmpdir,
+    cwd_tmpdir,
+    text="""# %%
+def func():
+    return 42
+# %%
+func()""",
+):
+    tmpdir.join("notebook.py").write(text)
+    jupytext(["--set-formats", "ipynb,py:percent", "notebook.py"])
+
+    nb = read(tmpdir.join("notebook.ipynb"))
+    assert nb.cells[0].metadata["lines_to_next_cell"] == 0
+
+    jupytext(["--sync", "notebook.py", "--pipe", "black"])
+    nb = read(tmpdir.join("notebook.ipynb"))
+    assert "lines_to_next_cell" not in nb.cells[0].metadata
+
+    new_text = tmpdir.join("notebook.py").read()
+    assert "\n\n# %%\nfunc()" in new_text

--- a/tests/test_cli.py
+++ b/tests/test_cli.py
@@ -867,7 +867,7 @@ def test_remove_jupytext_metadata(tmpdir, cwd_tmpdir):
 )
 def test_convert_and_update_preserves_notebook(nb_file, fmt, tmpdir, cwd_tmpdir):
     # cannot encode magic parameters in markdown yet
-    if "magic" in nb_file and fmt == "md":
+    if ("magic" in nb_file or "LateX" in nb_file) and fmt == "md":
         return
 
     tmp_ipynb = "notebook.ipynb"

